modules/ROOT/pages/mule-server-notification.adoc

```diff
@@ -0,0 +1,284 @@
+= Mule Server Notifications
+
+ifndef::env-site,env-github[]
+
+include::_attributes.adoc[]
+
+endif::[]
+
+:keywords: server notifications, log, downtime, monitor
+
+Mule provides an internal notification mechanism that you can use to access changes that occur on the Mule Server, such as adding a flow component, a request for authorization failing, or Mule starting. You can set up your agents or flow components to react to these notifications.
+
+== Configuring Notifications
+
+Message notifications provide a snapshot of all information sent into and out of the Mule Server. Mule fire these notifications whenever  it receives or sends a message. These additional notifications have some impact on performance, so they are disabled by default. To enable message notifications, you set the type of messages you want to enable using the `<notifications>` element in your Mule configuration file. You also register the notification listeners and associate interfaces with specific notifications.
+
+For example, first you create beans for the notification listeners in your spring configuration file, specifying the class of the type of notification you want to receive:
+
+[source,xml,linenums]
+
+----
+<bean name="notificationLogger" class="org.myfirm.ProcessorNotificationStore"/>
+----
+
+And then, add the reference for your spring configuration:
+
+[source,xml,linenums]
+
+----
+<spring:config name="springConfig" files="notification-beans.xml"/>
+----
+
+Next, you specify the notifications you want to receive using the `<notification>` element, and then register the listeners using the `<notification-listener>` element:
+
+[source,xml,linenums]
+
+----
+<notifications>
+    <notification event="MESSAGE-PROCESSOR"/>
+    <notification-listener ref="notificationLogger"/>
+</notifications>
+----
+
+When you specify the MESSAGE-PROCESSOR notification, a notification is sent before and after a message processor is invoked. Because the listeners implement the interface for the type of notification they want to receive, the listeners receive the correct notifications.
+
+For example, the `ProcessorNotificationLogger` class would implement `org.mule.runtime.api.notification.MessageProcessorNotificationListener`:
+
+[source,java,linenums]
+
+----
+public class ProcessorNotificationLogger implements MessageProcessorNotificationListener<MessageProcessorNotification> {
+
+    @Override
+    public void onNotification(MessageProcessorNotification notification) {
+        // TODO Auto-generated method stub
+    }
+}
+----
+
+[TIP]
+
+Don't forget to add the mule-api dependency in your pom file with provided as scope.
```

Comment:
I would explain why, "so that the api classes are available in your classpath. The depedency will then be provided by mule in runtime" or something like that.